From c5552442949fd5130323b7f653e1843948e57126 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Wed, 25 Nov 2020 19:58:25 +0100
Subject: [PATCH 1/4] serial-parsers: also match OK responses that are not at
 end of line

Do not fail to detect a valid response with a call or text incoming.
This happens also during port probing when there's no URC parsers
installed in the serial port. This probably will not happen when the
serial port was managed by the modem object.
---
 src/mm-serial-parsers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mm-serial-parsers.c b/src/mm-serial-parsers.c
index b511302e..248cc685 100644
--- a/src/mm-serial-parsers.c
+++ b/src/mm-serial-parsers.c
@@ -108,7 +108,7 @@ mm_serial_parser_v1_new (void)
 
     parser = g_slice_new (MMSerialParserV1);
 
-    parser->regex_ok = g_regex_new ("\\r\\nOK(\\r\\n)+$", flags, 0, NULL);
+    parser->regex_ok = g_regex_new ("\\r\\nOK(\\r\\n)+", flags, 0, NULL);
     parser->regex_connect = g_regex_new ("\\r\\nCONNECT.*\\r\\n", flags, 0, NULL);
     parser->regex_sms = g_regex_new ("\\r\\n>\\s*$", flags, 0, NULL);
     parser->regex_cme_error = g_regex_new ("\\r\\n\\+CME ERROR:\\s*(\\d+)\\r\\n$", flags, 0, NULL);
-- 
2.29.2

